lua/neo-tree/sources/common/components.lua

```diff
@@ -147,4 +148,71 @@ M.name = function(config, node, state)
   }
 end
 
+local indent_cache = {}
+
+local indent_without_markers = function(config, node, state)
+  local level = node.level
+  local indent_size = state.indent_size or 2
+
+  if level == 0 or indent_size == 0 then
+    return {}
+  end
+
+  if not indent_cache[level] then
+    local indent = utils.table_copy(indent_cache[level - 1] or {})
+
+    local spaces = utils.make_spaces(indent_size)
+    table.insert(indent, { text = spaces })
+
+    indent_cache[level] = indent
+  end
+
+  local node_indent = utils.table_copy(indent_cache[level])
+  -- table.insert(node_indent, { text = " " })
+
+  return node_indent
+end
+
+local prevent_indent = {}
+
+M.indent = function(config, node, state)
+  local indent_size = state.indent_size or 2
+  if indent_size == 0 then
+    return {}
+  end
+
+  local level = node.level
+  local with_markers = config.with_markers
+
+  if level < 2 or not with_markers then
+    return indent_without_markers(config, node, state)
```

Comment:
Actually, the entire function can probably be dropped and replaced with:
```suggestion
    return {
      text = string.rep(" ", indent_size * level),
      highlight = config.highlight or highlights.INDENT_MARKER
    }
```